### PR TITLE
Add web page endpoints for youtube channels

### PR DESCRIPTION
**Pull request by orthur**
Added `/yt/channels` and `/yt/channel/channelID` endpoints for web UI

localhost_8087_yt_channel_UCTVk323gzizpujtn2T_BL7w (1)

localhost_8087_yt_channels

### CODE REVIEW

File: app/youtube/feed/feed.go [hunk 1]
```diff
@@ -104,8 +104,9 @@ type Entry struct {
 		URI  string `xml:"uri"`
 	} `xml:"author"`
 
-	File     string
-	Duration int // seconds
+	File        string
+	Duration    int    // seconds
+	DurationFmt string // used for ui only in
```
Comment (umputun) [inline app/youtube/feed/feed.go:109]: looks like you have a part of the comment missing

**Review by umputun — COMMENTED** [review]
thx, minor comment iussue only

**Comment by umputun** [comment]
confused a little bit. isn't this the same thing as the current /sources, i.e. https://feedmaster.umputun.com/feed/echo-msk/sources

**Comment by orthur** [comment]
> confused a little bit. isn't this the same thing as the current /sources, i.e. https://feedmaster.umputun.com/feed/echo-msk/sources
> 
> upd: I meant the second picture

`/feed/feedName/sources` gets sources from `feeds:` field in `fm.yml`, but `/yt/channels` gets channels from `youtube` field

**Comment by umputun** [comment]
but why we even care? channels from youtube struct don't represent the actual output, but rathee yt sources used to make feeds defineds in feeds.<name>.sources. The part user see is from this feeds section, not from youtube section

**Comment by orthur** [comment]
I thought youtube section is separate thing for getting youtube audio and feeds section for other RSS podcasts. I use it without feeds section, hence I made web page to see which channels have set in youtube section

**Comment by umputun** [comment]
well, youtube part is making feeds as well, however this is done just to provide sources for the primary section. The intended use case can be seen in the provided https://github.com/umputun/feed-master/blob/master/yt-test.yml 

As you can see feeds.yt-example.sources just using youtube.channels.

**Comment by orthur** [comment]
Yes I got it. I am fine with closing this PR if changes is not needed

**Comment by umputun** [comment]
I think the part of UI to show the actual list of items per channel is a useful one. It will be great if it can be adapted to work with sources. We also can replace the link from /sources to this one

**Comment by orthur** [comment]
what do you mean to "work with sources"?

**Comment by umputun** [comment]
Instead of youtube-specific /yt/channel/channelID the page should be for any source, i.e. smth like /feed/{name}/source/{src} which show the same page, but instead of making it from yt channel info, render it from the higher level feed info (i.e. `s.Conf.Feeds`)

**Comment by umputun** [comment]
> what would be `{src}` value? `id: UCWAIvx2yYLK_xTYD4F2mUNw`?

the src is `feeds.{name}.sources.name` value, i.e. for the provided conf, it would be `Точка`, `Живой Гвоздь`, and so on

```yml
feeds:
  yt-example:
    title: Some cool channels
    description: an example of youtube-based podcast
    link: http://example.com
    language: "ru-ru"
    image: images/yt-example.png
    telegram_channel: udev_test
    sources:
      - name: Точка
        url: http://localhost:8080/yt/rss/PLZVQqcKxEn_6YaOniJmxATjODSVUbbMkd
      - name: Живой Гвоздь
        url: http://localhost:8080/yt/rss/UCWAIvx2yYLK_xTYD4F2mUNw
      - name: Дилетант
        url: http://localhost:8080/yt/rss/UCuIE7-5QzeAR6EdZXwDRwuQ
      - name: Сергей Асланян
        url: http://localhost:8080/yt/rss/UCPemWOA2UiBwhTauetmz95w
```

**Comment by umputun** [comment]
note: you  last merge/rebase made this PR very confusing. I suggest to replace it on top of the current master or somehow squash. @paskal may know what git magic needed for this

**Comment by paskal** [comment]
`git rebase upstream/master` is what you need to do in your branch.

**Comment by orthur** [comment]
changed endpoint to `/feed/name/source/source`

**Review by umputun — CHANGES_REQUESTED** [review]
the link from /sources still points to the old place, it should be changed to the new url (i.e. source/blah) you have added

**Review by umputun — COMMENTED** [review]
LGTM. I would like to merge it but pls rebase it first and make a single (or a pair) of committs. If i squash it with github it will rewrite commit and set me as the author

**Pull request merged**